app/Repositories/BatchRepository.php

```diff
@@ -0,0 +1,17 @@
+<?php
+
+namespace App\Repositories;
+
+use App\Helper\Constant;
+use App\Models\Batch;
+
+class BatchRepository extends CrudRepository
+{
+    protected string $model = Batch::class;
+
+    public function deleteBatch($batch): Batch
+    {
+        $batch->update(['deleted' => Constant::SOFT_DELETE]);
+        return $batch;
+    }
```

Comment:
I would prefar move this to `CrudRepository` and name somthing like - `softDeleteItem()`

and maybe we'll there create other method for `hardDeleteItem()`


And then remove it from here.

Reply:
solved